src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -80,9 +82,11 @@ private void registerConnectionsGauge() {
     }
   }
 
-  protected JmxProxy connectImpl(Node node, int connectionTimeout) throws ReaperException, InterruptedException {
+  protected JmxProxy connectImpl(Node node, AppContext context) throws ReaperException, InterruptedException {
```

Comment:
`AppContext` can be injected in via the constructor, rather than every method.

Reply:
Done.

We have a circular reference though as the jmxConnectionFactory now takes the context as argument and is stored in the same context.
Not sure that's really a problem though.